src/v/archival/ntp_archiver_service.cc

```diff
@@ -538,6 +555,17 @@ ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
           "Failed to download partition manifest in read-replica mode");
         co_return res;
     } else {
+        if (m == _parent.archival_meta_stm()->manifest()) {
```

Comment:
There was some earlier discussion about potentially needing to omit the insync offset. Looking back, I don't imagine that actually having any repercussions so perhaps it's fine as is. Just wanted to call it out that we're intentionally ignoring those fields, right?

Reply:
Right, they're effectively meaningless on a read replica, because we're never using them for max collectible offset.  This would be something to pick up as/when we ever implemented a "transition to primary" for read replicas.